grease/src/Grease/Shape/Pointer.hs

```diff
@@ -238,7 +246,7 @@ initializeMemShape tag =
 --
 -- * @wptr@: Width of a pointer, in bits
 -- * @tag@: See 'Grease.Shape.Shape'
-newtype PtrTarget wptr tag = PtrTarget (Seq (MemShape wptr tag))
+data PtrTarget wptr tag = PtrTarget (Maybe BlockId) (Seq (MemShape wptr tag)) 
```

Comment:
Do we expect that any two `PtrTarget`s with the same non-`Nothing` block ID will also have the same `Seq` of `MemShape`s? Would be good to document the expectation in either case.